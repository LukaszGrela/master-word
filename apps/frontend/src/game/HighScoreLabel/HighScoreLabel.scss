@import '../../styles/variables';

.highscore-label {
  background-color: $results-highscore-bgcolor;
  padding: 0.25rem 0.5rem;
  max-width: 6rem;
  display: block;
  border: 0.25rem $results-highscore-color solid;
  border-radius: 0.5rem;
  color: $results-highscore-color;
  font-weight: 600;
  position: absolute;
  right: 0;
  top: 0;
  transform: rotate(0.08turn);
  box-shadow: 1px 1px 7px 0px black;

  @media (prefers-color-scheme: dark) {
    color: $results-highscore-color-dark;
    border-color: $results-highscore-color-dark;

    text-shadow: 2px 2px 3px black;
  }
}
