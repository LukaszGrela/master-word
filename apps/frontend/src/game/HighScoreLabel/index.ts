/* v8 ignore start */
export { default as HighScoreLabel } from './HighScoreLabel';
/* v8 ignore stop */
