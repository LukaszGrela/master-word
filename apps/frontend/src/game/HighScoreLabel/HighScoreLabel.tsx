import { FC } from 'react';
import { useLanguage } from '../../i18n';
import './HighScoreLabel.scss';

const HighScoreLabel: FC = (): JSX.Element => {
  const { getUIText: t } = useLanguage();

  return <span className="highscore-label">{t('highscore-label')}</span>;
};

export default HighScoreLabel;
