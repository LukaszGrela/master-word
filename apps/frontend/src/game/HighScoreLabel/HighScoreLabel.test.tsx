import { describe, expect, it, vi } from 'vitest';
import {
  contextRenderer,
  defaultLanguageContext,
} from '../../i18n/LanguageContext/__tests__/contextRenderer';

import HighScoreLabel from './HighScoreLabel';

describe('game', () => {
  describe('HighScoreLabel', () => {
    it('Renders with no props', () => {
      const { container } = contextRenderer(<HighScoreLabel />);

      expect(container).toMatchSnapshot();
    });
    it('Renders with context', () => {
      const { container } = contextRenderer(<HighScoreLabel />, {
        languageContext: {
          ...defaultLanguageContext,
          getUIText: vi.fn().mockReturnValue('New Highscore'),
        },
        renderOptions: {},
      });

      expect(container).toMatchSnapshot();
    });
  });
});
