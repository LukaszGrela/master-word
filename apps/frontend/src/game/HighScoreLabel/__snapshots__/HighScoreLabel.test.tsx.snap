// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > HighScoreLabel > Renders with context 1`] = `
<div>
  <span
    class="highscore-label"
  >
    New Highscore
  </span>
</div>
`;

exports[`game > HighScoreLabel > Renders with no props 1`] = `
<div>
  <span
    class="highscore-label"
  />
</div>
`;
